transcendence-app/src/user/user.service.ts

```diff
@@ -84,11 +86,37 @@ export class UserService {
     );
   }
 
-  updateUser(
+  async updateUser(
     userId: string,
     updateUserDto: UpdateUserDto,
   ): Promise<User | null> {
-    return this.userRepository.updateById(userId, updateUserDto);
+    if (!updateUserDto.oldPassword && !updateUserDto.password) {
+      return this.userRepository.updateById(userId, updateUserDto);
+    }
+    const { oldPassword, confirmationPassword, ...updateUserPassword } =
+      updateUserDto;
```

Comment:
praise: Nice implementation :+1: 

suggestion: We could check if the password matches the confirmation password.

Reply:
Thanks, It's done in the front-end but yes we can do it also here